src/test/java/org/opensearch/security/authtoken/jwt/JwtVendorTest.java

```diff
@@ -0,0 +1,80 @@
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+package org.opensearch.security.authtoken.jwt;
+
+import java.util.function.LongSupplier;
+
+import org.apache.cxf.rs.security.jose.jwk.JsonWebKey;
+import org.apache.cxf.rs.security.jose.jws.JwsJwtCompactConsumer;
+import org.apache.cxf.rs.security.jose.jwt.JwtToken;
+import org.junit.Assert;
+import org.junit.Test;
+
+import org.opensearch.common.settings.Settings;
+
+public class JwtVendorTest {
+
+    @Test
+    public void testCreateJwkFromSettings() throws Exception {
+        Settings settings = Settings.builder()
+                .put("signing_key", "abc123").build();
+
+        JsonWebKey jwk = JwtVendor.createJwkFromSettings(settings);
+        Assert.assertEquals("HS512", jwk.getAlgorithm());
+        Assert.assertEquals("sig", jwk.getPublicKeyUse().toString());
+        Assert.assertEquals("abc123", jwk.getProperty("k"));
+    }
+
+    @Test (expected = Exception.class)
+    public void testCreateJwkFromSettingsWithoutSigningKey() throws Exception{
+        Settings settings = Settings.builder()
+                .put("jwt", "").build();
+        JwtVendor.createJwkFromSettings(settings);
+    }
+
+    @Test
+    public void testCreateJwt() throws Exception {
+        String issuer = "cluster_0";
+        String subject = "admin";
+        String audience = "extension_0";
+        Integer expirySeconds = 300;
+        LongSupplier currentTime = () -> (int)100;
+        Settings settings =  Settings.builder().put("signing_key", "abc123").build();
+        Long expectedExp = currentTime.getAsLong() + (expirySeconds * 1000);
+
+        JwtVendor jwtVendor = new JwtVendor(settings, currentTime);
+        String encodedJwt = jwtVendor.createJwt(issuer, subject, audience, expirySeconds);
+
+        JwsJwtCompactConsumer jwtConsumer = new JwsJwtCompactConsumer(encodedJwt);
+        JwtToken jwt = jwtConsumer.getJwtToken();
+
+        Assert.assertEquals("cluster_0", jwt.getClaim("iss"));
+        Assert.assertEquals("admin", jwt.getClaim("sub"));
+        Assert.assertEquals("extension_0", jwt.getClaim("aud"));
+        Assert.assertNotNull(jwt.getClaim("iat"));
+        Assert.assertNotNull(jwt.getClaim("exp"));
+        Assert.assertEquals(expectedExp, jwt.getClaim("exp"));
+    }
+
+    @Test (expected = Exception.class)
+    public void testCreateJwtWithBadExpiry() throws Exception {
+        String issuer = "cluster_0";
+        String subject = "admin";
+        String audience = "extension_0";
+        Integer expirySeconds = -300;
+
+        Settings settings =  Settings.builder().put("signing_key", "abc123").build();
+        JwtVendor jwtVendor = new JwtVendor(settings);
+
+        jwtVendor.createJwt(issuer, subject, audience, expirySeconds);
+    }
+}
```

Comment:
Would do you think about adding a test for a bad not before time? That is a common part of JWTs as I understand so we may consider having a test for the scenario. That being said it does not look like you ever set a not before time on creation so perhaps that is why you did not test it.

Do you think we should add a not before time or no? It is basically a marker for when the token becomes live. 

Overall great work on this! I think it is in a good spot and if you could add the comment about the default values and address Peter's other comment your thumbs-up'd I would be ready to approve.

Great work Ryan!

Reply:
That's actually a good one. Because for the current design `iat` and `nbf` should be the same. I'm pinging @cwperks and @peternied here to ask that, is there any extended usage of the claim `nbf`?

Reply:
@scrawfor99 After talking with @cwperks, I think the original purpose of this `nbf` claim is for issue a token for future usage, which is not our case, at least for now. But, I think it is ok to have it here, so that we can also have some test for the authentication backend to check. Still, we can always go back and change it later.